lib/get/getCachePosition.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * getCachePosition makes a fast walk to the bound value since all bound
+ * paths are the most possible optimized path.
+ *
+ * @param {Model} model -
+ * @param {Array} path -
+ * @returns {Mixed} - undefined if there is nothing in this position.
+ * @private
+ */
+module.exports = function getCachePosition(model, path) {
+    var currentCachePosition = model._root.cache;
+    var depth = -1;
+    var maxDepth = path.length;
+
+    // The loop is simple now, we follow the current cache position until
+    //
+    while (++depth < maxDepth &&
+           currentCachePosition && !currentCachePosition.$type) {
+
+        currentCachePosition = currentCachePosition[path[depth]];
+    }
+
+    return [
```

Comment:
Do we need to return an array here? Expensive plus not very ergonomic. Can't we return `(null || currentCachePosition)` - or some other constant to reflect SHORTED. As far as I can tell, the consuming code doesn't need the currentCachePosition for the SHORTED case, and the CACHE_POSITION case doesn't need to know we're shorter.